online/src/main/scala/ai/chronon/online/FetcherBase.scala

```diff
@@ -108,13 +108,22 @@ class FetcherBase(kvStore: KVStore,
                          totalResponseValueBytes)
 
         val batchIr = toBatchIr(batchBytes, servingInfo)
+        // todo: handle group by schema evolution for tiled ir and streaming rows
         val output: Array[Any] = if (servingInfo.isTilingEnabled) {
           val streamingIrs: Iterator[TiledIr] = streamingResponses.iterator
             .filter(tVal => tVal.millis >= servingInfo.batchEndTsMillis)
-            .map { tVal =>
-              val (tile, _) = servingInfo.tiledCodec.decodeTileIr(tVal.bytes)
-              TiledIr(tVal.millis, tile)
+            .flatMap { tVal =>
+              Try(servingInfo.tiledCodec.decodeTileIr(tVal.bytes)) match {
+                case Success((tile, _)) => Array(TiledIr(tVal.millis, tile))
+                case Failure(_) =>
+                  logger.error(
```

Comment:
either in this PR or a followup, lets add the ability to enable / disable this behavior.
We could use either the flag api or a per groupBy / join config setting

Reply:
Hi @piyushn-stripe , yes, I am not going to merge this PR as is. I will do a refactor on it.